controllers/provisioning_enabled.go

```diff
@@ -0,0 +1,60 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	osconfigv1 "github.com/openshift/api/config/v1"
+	osclientset "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/cluster-baremetal-operator/api/v1alpha1"
+)
+
+func IsEnabled(ctx context.Context, osClient osclientset.Interface) (bool, error) {
+	infra, err := osClient.ConfigV1().Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return false, errors.Wrap(err, "unable to determine Platform")
+	}
+
+	if infra.Status.ControlPlaneTopology == osconfigv1.ExternalTopologyMode {
+		return false, nil
+	}
+
+	switch infra.Status.Platform {
+	case osconfigv1.BareMetalPlatformType, osconfigv1.OpenStackPlatformType, osconfigv1.NonePlatformType, osconfigv1.VSpherePlatformType:
+		return true, nil
+	default:
+		return false, nil
+	}
+}
+
+func EnabledFeatures(ctx context.Context, osClient osclientset.Interface) (v1alpha1.EnabledFeatures, error) {
```

Comment:
Both these methods are reading the same resources. Even though they are read from the cache, we could make it more efficient by setting EnabledFeatures while determining if this is a supported platform.